doc/multi_asic/SONiC_multi_asic_hld.md

```diff
@@ -0,0 +1,937 @@
+  - ASIC_NAME: This field is the linux Namespace to which this ASIC belongs
+  - SUB_ROLE: This field indicates if ASIC is Frontend or Backend
+
+ These additional fields are not present on the device DEVICE_METADATA for single ASIC devices
+
+#### 2.4.3.3. Router Mac
+In a multi ASIC system, all the frontend ASICs will have the same router mac address. The frontend ASICs will use the system base mac as router mac.  
+Each backend ASIC will have different router Mac address. The mac address of `eth0`  in the namespace will be used as the router mac on the backend ASIC.
+
+#### 2.4.3.4. Loopback address
+The system will have 2 Loopback Interfaces
+
+Loopback0 has a globally unique IP address, which is advertised by the multi-ASIC device to its peers.
+This way all the external devices will see this device as a single device.
+
+Loopback4096 is assigned an IP address which has a scope within the device only. Each ASIC has a different ip address for Loopback4096. This ip address will be used as Router-Id by the bgp instance on multi ASIC devices.
+
+#### 2.4.3.5. Default Route
+
+For IPv4 there is default route added by docker when it comes in asic namespace which has higher priority over FRR/BGP learn default route because of BGP router does not get programmed. To overcome this
+BGP docker when comes in asic namespace will remove the default docker route and add it back with lower priority(higher metric value)
+
+For IPv6 sysctl.net.ipv6.conf.all.forwarding was 0 in asic namespace which makes default route learn on local link-address and not on global address. To overcome this and any other possible mismatch in sysctl.net.*
+value between host and asic namespace common file having these values will be used and applied for both host and asic namespace. 
+
+#### 2.4.3.6. BGP configuration
+ When BGP is running at back-end ASICs, some more bgp configurations are required. They are:
+  - Redistribution of connected routes are configured on frontend ASICs
+  - Route reflector is configured on the backend ASICs.
+  - All the internal neighbors are configured *next-hop-self self*
+    
+### 2.4.4. Database Service
+There will be  **N+1** database containers running in a **N**-ASIC device with a single CPU. 
+
+  -  The database container running in the linux host will be called _globalDB_ container in a multi asic platform. The database tables populated here will be system wide attributes like SYSLOG, AAA, TACACS, Mgmt interface and resources like fan, psu, thermal.
+  -  The  ***database{n}*** container will be started in each namespace. _n_  here denotes the ASIC_ID, These databases will be used and updated by applications in that namespace.
+
+Each database container will have it's own **"/var/run/redis{n}/"** directory which contains the database_config.json file, redis unix socket etc. The database_config.json will no longer be a static build time file.

```

Comment:
As SONiC point of view, why don't use single database container to configure multi-asic ? what is the rational behind several database containers to configure multi-asic system?